tests/foreman/cli/test_hammer.py

```diff
@@ -84,66 +85,69 @@ def __init__(self, *args, **kwargs):
-        command_subcommands = set(
-            [subcommand['name'] for subcommand in output['subcommands']]
-        )
-        if 'discovery_rule' in command and bz_bug_is_open(1219610):
-            # Adjust the discovery_rule subcommand name. The expected data is
-            # already with the naming convetion name
-            expected = _fetch_command_info(
-                command.replace('discovery_rule', 'discovery-rule'))
-        else:
-            expected = _fetch_command_info(command)
-        expected_options = set()
-        expected_subcommands = set()
-
-        if expected is not None:
-            expected_options = set(
-                [option['name'] for option in expected['options']]
+        raw_output = ssh.command(
+            'hammer full-help', output_format='plain').stdout
+        commands = re.split('.*\n(?=hammer.*\n^[-]+)', raw_output, flags=re.M)
+        commands.pop(0)  # remove "Hammer CLI help" line
+        for raw_command in commands:
+            raw_command = raw_command.splitlines()
+            command = raw_command.pop(0).replace(' >', '')
+            output = hammer.parse_help(raw_command)
+            command_options = set([
+                option['name'] for option in output['options']])
+            command_subcommands = set(
+                [subcommand['name'] for subcommand in output['subcommands']]
             )
-            expected_subcommands = set(
-                [subcommand['name'] for subcommand in expected['subcommands']]
-            )
-
-        if command == 'hammer' and bz_bug_is_open(1219610):
-            # Adjust the discovery_rule subcommand name
-            command_subcommands.discard('discovery_rule')
-            command_subcommands.add('discovery-rule')
-
-        added_options = tuple(command_options - expected_options)
-        removed_options = tuple(expected_options - command_options)
-        added_subcommands = tuple(command_subcommands - expected_subcommands)
-        removed_subcommands = tuple(expected_subcommands - command_subcommands)
-
-        if (added_options or added_subcommands or removed_options or
-                removed_subcommands):
-            diff = {
-                'added_command': expected is None,
-            }
-            if added_options:
-                diff['added_options'] = added_options
-            if removed_options:
-                diff['removed_options'] = removed_options
-            if added_subcommands:
-                diff['added_subcommands'] = added_subcommands
-            if removed_subcommands:
-                diff['removed_subcommands'] = removed_subcommands
-            self.differences[command] = diff
-
-        if len(output['subcommands']) > 0:
-            for subcommand in output['subcommands']:
-                self._traverse_command_tree(
-                    '{0} {1}'.format(command, subcommand['name'])
+            if 'discovery_rule' in command and bz_bug_is_open(1219610):
```

Comment:
Think we can remove this work around, the bug is closed want fix , and the command is as expected  discovery-rule for all 6.3 snaps (from the beginning)